sparse_strips/vello_bench/src/data.rs

```diff
@@ -0,0 +1,242 @@
+                .join("../../examples/assets/Ghostscript_Tiger.svg"),
+        ));
+
+        for entry in std::fs::read_dir(&data_dir).unwrap() {
+            let entry = entry.unwrap();
+            let path = entry.path();
+
+            if path.extension().and_then(|e| e.to_str()) == Some("svg") {
+                data.push(read(&path));
+            }
+        }
+
+        data
+    })
+}
+
+#[derive(Clone, Debug)]
+pub struct DataItem {
+    pub name: String,
+    pub fills: Vec<BezPath>,
+    pub strokes: Vec<BezPath>,
+    pub width: u16,
+    pub height: u16,
+}
+
+impl DataItem {
+    /// Get the raw flattened lines of both fills and strokes.
+    ///
+    /// A stroke width of 2.0 is assumed.
+    pub fn lines(&self) -> Vec<Line> {
+        let mut line_buf = vec![];
+        let mut temp_buf = vec![];
+
+        for path in &self.fills {
+            flatten::fill(path, Affine::default(), &mut temp_buf);
+            line_buf.extend(&temp_buf);
+        }
+
+        let stroke = Stroke {
+            // Obviously not all strokes have that width, but it should be good enough
+            // for benchmarking.
+            width: 2.0,
+            ..Default::default()
+        };
+
+        for path in &self.strokes {
+            flatten::stroke(path, &stroke, Affine::default(), &mut temp_buf);
+            line_buf.extend(&temp_buf);
+        }
+
+        line_buf
+    }
+
+    /// Get the unsorted tiles.
+    pub fn unsorted_tiles(&self) -> Tiles {
+        let mut tiles = Tiles::new();
+        let lines = self.lines();
+        tiles.make_tiles(&lines, self.width, self.height);
+
+        tiles
+    }
+
+    /// Get the sorted tiles.
+    pub fn sorted_tiles(&self) -> Tiles {
+        let mut tiles = self.unsorted_tiles();
+        tiles.sort_tiles();
+
+        tiles
+    }
+
+    /// Get the alpha buffer and rendered strips.
+    pub fn strips(&self) -> (Vec<u8>, Vec<Strip>) {
+        let mut strip_buf = vec![];
+        let mut alpha_buf = vec![];
+        let lines = self.lines();
+        let tiles = self.sorted_tiles();
+
+        strip::render(
+            &tiles,
+            &mut strip_buf,
+            &mut alpha_buf,
+            Fill::NonZero,
+            &lines,
+        );
+
+        (alpha_buf, strip_buf)
+    }
+}
+
+fn read(path: &Path) -> DataItem {
+    let file_name = {
+        let tmp = path.with_extension("");
+        tmp.file_name().unwrap().to_string_lossy().to_string()
+    };
+
+    let data = std::fs::read(path).unwrap();
+    let tree = usvg::Tree::from_data(&data, &usvg::Options::default()).unwrap();
+    let mut ctx = ConversionContext::new();
+    convert(&mut ctx, tree.root());
+
+    DataItem {
+        name: file_name,
+        fills: ctx.fills,
+        strokes: ctx.strokes,
+        #[allow(
+            clippy::cast_possible_truncation,
+            reason = "It's okay to ignore for benchmarking."
+        )]
+        width: tree.size().width() as u16,
```

Comment:
I'd probably just use `try_into().expect()` here; if benchmarking is rendering into an image 65000 pixels too small, we probably want to know about it

Reply:
(The input is a float, which doesn't support tryinto, this is https://github.com/rust-lang/rfcs/issues/3304)